scripts/repo-variable-writer.go

```diff
@@ -0,0 +1,373 @@
+	return strings.ToLower(projName + "_version")
+}
+
+type githubApiEndpointType string
+
+const (
+	ghRepoVariableEndpoint     githubApiEndpointType = "https://api.github.com/repos/%s/actions/variables/%s"
+	ghWorkflowDispatchEndpoint githubApiEndpointType = "https://api.github.com/repos/%s/actions/workflows/%s/dispatches"
+)
+
+func (obj *GithubVariable) genUrlAndHeaders(
+	endpointType githubApiEndpointType,
+	variableName string, workflowFileName string) (*string, map[string]string, error) {
+
+	url := ""
+
+	switch endpointType {
+	case ghRepoVariableEndpoint:
+		url = fmt.Sprintf(
+			string(endpointType),
+			obj.fullRepoName,
+			variableName,
+		)
+	case ghWorkflowDispatchEndpoint:
+		url = fmt.Sprintf(
+			string(endpointType),
+			obj.fullRepoName,
+			workflowFileName,
+		)
+	}
+
+	return toPtr[string](url),
+		map[string]string{
+			"Accept":               "application/vnd.github+json",
+			"Authorization":        "Bearer " + obj.githubToken,
+			"X-GitHub-Api-Version": "2022-11-28",
+		}, nil
+}
+
+func (obj *GithubVariable) UpdateRepoVariable(projName, newVersion string) error {
+
+	variableName := obj.genVariableName(projName)
+
+	url, header, err := obj.genUrlAndHeaders(ghRepoVariableEndpoint, variableName, "")
+	if err != nil {
+		return err
+	}
+
+	newVariableData := struct {
+		Name string `json:"name"`
+		Val  string `json:"value"`
+	}{
+		Name: strings.ToUpper(variableName),
+		Val:  newVersion,
+	}
+
+	var _newVariableData bytes.Buffer
+
+	if err := json.NewEncoder(&_newVariableData).Encode(newVariableData); err != nil {
+		return fmt.Errorf("failed to serialize the body: %v", err)
+	}
+
+	resp, err := handleHTTPCall(
+		http.MethodPatch,
+		*url,
+		time.Minute,
+		&_newVariableData,
+		header,
+	)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("status code was not 204, got: %v", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func (obj *GithubVariable) ReadRepoVariable(projName string) (*string, error) {
+
+	variableName := obj.genVariableName(projName)
+	url, header, err := obj.genUrlAndHeaders(ghRepoVariableEndpoint, variableName, "")
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := handleHTTPCall(
+		http.MethodGet,
+		*url,
+		time.Minute,
+		nil,
+		header,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("status code was not 200, got: %v", resp.StatusCode)
+	}
+
+	var variableData struct {
+		VariableValue string `json:"value"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&variableData); err != nil {
+		return nil, fmt.Errorf("failed deserialize response body: %v", err)
+	}
+
+	return toPtr[string](variableData.VariableValue), nil
+}
+
+func (obj *GithubVariable) UpdatedProjectDispatcher(projData Project, version string) error {
+
+	projToWorkflowTable := map[string]string{
+		"falco": "falco.yaml",
```

Comment:
this probably should not get hardcoded

Reply:
yes currently wait for ross to look at this and I can then update the logic here

like do we have different workflow files or a single workflow file like in #105

Reply:
> yes currently wait for ross to look at this and I can then update the logic here

> like do we have different workflow files or a single workflow file like in https://github.com/cncf-tags/green-reviews-tooling/pull/105

@dipankardas011 There will be a single workflow file. The project name is passed to the workflow in the `cncf_project` input.